INN/umbrella-energynewsnetwork
# Digest search functionality

## Changes

This pull request makes the following changes:

- [x] custom template for the Digest category
- [x] search box
    - [x] search box
    - [x] filter buttons for region
    - [x] datepicker input
- [x] search filtering
    - [x] by words passed in the search box
    - [x] by the region-specific checkboxes
    - [x] by a date picker
- [x] ~~also affecting LMP?~~ Remove LMP, increase query size to 20.

Form in Firefox:
![Screen Shot 2020-02-10 at 13 17 13](https://user-images.githubusercontent.com/1754187/74178327-6029f300-4c09-11ea-94bf-bd35023db9ed.png)

Firefox's datepicker
![Screen Shot 2020-02-10 at 13 27 24](https://user-images.githubusercontent.com/1754187/74178340-66b86a80-4c09-11ea-9af0-cbc73b4739c7.png)

Inputting information in Safari, which lacks datepicker support (https://bugs.webkit.org/show_bug.cgi?id=119175 and https://caniuse.com/#feat=input-datetime)
![Screen Shot 2020-02-10 at 13 24 37](https://user-images.githubusercontent.com/1754187/74178356-7172ff80-4c09-11ea-8342-cd608b288fb0.png)
![Screen Shot 2020-02-10 at 13 27 12](https://user-images.githubusercontent.com/1754187/74178358-7172ff80-4c09-11ea-9f08-6eb29a2b6069.png)



## Why



For #77

## Testing/Questions

Features that this PR affects:

- `/category/digests/`


Questions that need to be answered before merging:

- [x] do the date filters work? I did some testing and it appeared to work
- [ ] do we worry about the paradoxical situation where the "before" date is before the "after" date?
- [ ] does this need a polyfill datepicker for Safari or other browsers that don't have a native date picker for `input type="date"`? Safari supports input validation, so we can use `:invalid` as a selector to reveal additional information, as this PR does.

Steps to test this PR:

1. check out the PR, using a recent DB from production
2. go to `/category/digests/`
3. perform various searches:
    - by word
    - by date
    - by region
    - by any combination thereof
4. Examine each search's results to determine whether the returned posts match the query
